cli/types/index.d.ts

```diff
@@ -720,6 +720,8 @@ declare namespace Cypress {
      *
      * @see https://on.cypress.io/then
      */
+    then<S extends object | any[] | string | number | boolean>(fn: (this: ObjectLike, currentSubject: Subject) => Chainable<S>, options?: Partial<Timeoutable>): Chainable<S>
+    then<S extends object | any[] | string | number | boolean>(fn: (this: ObjectLike, currentSubject: Subject) => PromiseLike<S>, options?: Partial<Timeoutable>): Chainable<S>

```

Comment:
An option was coming up with a custom `Thenable` interface to match both uses cases, but I figured I'd just use standard ones (we define `Chainable` and TypeScript defines `PromiseLike` for all Spec compliant Promise implementations (including Bluebird)).